Source/WebCore/platform/graphics/ImageUtilities.h

```diff
@@ -44,5 +47,8 @@ WEBCORE_EXPORT Vector<String> findImagesForTranscoding(const Vector<String>& pat
 // happens while transcoding, a null string will be added to the returned list.
 WEBCORE_EXPORT Vector<String> transcodeImages(const Vector<String>& paths, const String& destinationUTI, const String& destinationExtension);
 
+WEBCORE_EXPORT std::pair<String, Vector<IntSize>> utiAndAvailableSizesFromImageData(std::span<const uint8_t>);
```

Comment:
Should not it return an `std::optional<std::pair<String, Vector<IntSize>>>` so it can return `std::nullopt` if an error happens?

We can also return 
```
enum class DecodingError {
    None,
    DataError,
    UnsupportedType
};

Expected<std::pair<String, Vector<IntSize>, DecodingError>>
```

Reply:
Updated return value to `Expected<std::pair<String, Vector<IntSize>, ImageDecodingError>>`